File: solucoes/queue/copa-do-mundo/gabrielbergoc/copa.cpp
#include <bits/stdc++.h>

using namespace std;

typedef pair<int, int> Jogo;
typedef char Time;

int main() {
    queue<Time> times;
    for (Time c = 'A'; c < 'P' + 1; c++) {
        times.push(c);
    }

    queue<Jogo> jogos;
    for (int x, y, i = 0; i < 15; i++) {
        cin >> x >> y;
        jogos.emplace(x, y);
    }

    while (!jogos.empty()) {
        Time time_1 = times.front();
        times.pop();
        
        Time time_2 = times.front();
        times.pop();

        int gols_1 = jogos.front().first;
        int gols_2 = jogos.front().second;
        jogos.pop();

        Time vencedor = gols_1 > gols_2 ? time_1 : time_2;
        times.push(vencedor);
    }

    cout << times.front() << endl;
}

File: solucoes/queue/copa-do-mundo/gabrielbergoc/copa.py
class Fila:
    def __init__(self, fila = None):
        if fila is None:
            self.__fila = []
        else:
            self.__fila = fila
        
    def push(self, value):
        self.__fila.append(value)
        
    def pop(self):
        return self.__fila.pop(0)

    def front(self):
        return self.__fila[0]

    def is_empty(self):
        return len(self.__fila) == 0


jogos = Fila([input() for _ in range(15)])
times = Fila([chr(c) for c in range(ord('A'), ord('P') + 1)])

while not jogos.is_empty():
    time_1, time_2 = times.pop(), times.pop()
    gols_1, gols_2 = (int(gols) for gols in jogos.pop().split(' '))
    vencedor = time_1 if gols_1 > gols_2 else time_2
    times.push(vencedor)

print(times.front())
